install/management/appliance.go

```diff
@@ -348,12 +330,19 @@ func (d *Dispatcher) setMacToGuestInfo() error {
 			err = errors.Errorf("Timeout to get VM MAC address")
 			return err
 		}
-		netName := d.nics[key]
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/portgroup", netName), Value: key})
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/mac", netName), Value: value})
-		keys = append(keys, netName)
+		for nicName, netInfo := range conf.Networks {
```

Comment:
This entire nested for loop construct can be replaced with the following - you should never need nested loops like this to discover if a key is in a hashmap:
```
	for name, info := range conf.Networks {
		value, ok := m[info.PortGroupName]
		if !ok || value == "" {
			// timeout to wait MAC address, so empty mac address is returned
			return errors.Errorf("Timeout getting MAC address for %s on %s", name, info.PortGroupName)
		}

		info.Mac = value
	}
```

I've left out the extraconfig manipulation from that loop for clarity.